src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -1311,11 +1311,19 @@ export class BundleDataClient {
             "fastFillsReplacingSlowFills should contain only deposits that can be slow filled"
           );
           const destinationBlockRange = getBlockRangeForChain(blockRangesForChains, destinationChainId, chainIds);
+          const originBlockRange = getBlockRangeForChain(blockRangesForChains, originChainId, chainIds);
+          const matchedDeposit = deposits[0];
+          // For a slow fill leaf to have been created (and subsequently create an excess), the deposit matching the
+          // slow fill request must have been included in a previous bundle AND the slow fill request should not
+          // be included in this bundle. This is because a slow fill request is only valid once its deposit
+          // has been mined, so if the deposit is not in a prior bundle to the fill, then no slow fill leaf could
+          // have been created. Secondly, the slow fill request itself must have occurred in an older bundle than the
+          // fill otherwise the slow fill leaf be unexecutable for an already-filled deposit..
           if (
+            matchedDeposit.blockNumber < originBlockRange[0] &&
             // If there is a slow fill request in this bundle that matches the relay hash, then there was no slow fill
             // created that would be considered excess.
-            !slowFillRequest ||
-            slowFillRequest.blockNumber < destinationBlockRange[0]
+            (!slowFillRequest || slowFillRequest.blockNumber < destinationBlockRange[0])
```

Comment:
This is only tangential because this was existing implementation, but we could optionally make this more deterministic by identifying the block where the slow fill request occurred. We don't currently have a utility for this but could add one without much effort (maybe as a follow-up). This would be a bit more robust than just reacting to no event, because that could occur due to an RPC issue.

Reply:
i don't like this, the bundle data client assumes that all events it gets are correct. If we make an exception and here and make an additional RPC query then we'd want to do the same in many places in the client. This would change the code a lot and make it more robust yes, but also much more expensive to run. I think its an absolutely fair assumption that the spoke pool clients have valid data